### PR TITLE
CDAP-3574 Fix session start in HDP Hive 0.12

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA - https://issues.cask.co/browse/CDAP-3574

### CODE REVIEW

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-DUT2720-1

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java [hunk 1]
```diff
@@ -129,4 +135,29 @@ protected OperationHandle doExecute(SessionHandle sessionHandle, String statemen
...
+
+    // HDP Hive 12 only sets SessionState when an operation is executed.
+    // The SessionState is not set during an openSession call.
+    // Hence execute a dummy statement during openSession, so that SessionState gets set.
+    OperationHandle opHandle = null;
+    try {
+      opHandle = getCliService().executeStatement(sessionHandle, "set cdap_tmp_var=1",
+                                                  Collections.<String, String>emptyMap());
+    } catch (Throwable t) {
+      LOG.error("Error setting temporary variable to set session state", t);
```
Comment (bdmogal) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java]: is it ok to ignore the exception? Do we presume that it even if there is an exception, the ``SessionHandle`` is set?

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java [hunk 2]
```diff
@@ -1156,17 +1156,6 @@ private boolean requiresUpgrade(TableInfo tableInfo) {
     return false;
   }
 
-  private SessionHandle openSession(Map<String, String> sessionConf) throws HiveSQLException {
```
Comment (bdmogal) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: is this unused?

Comment (bdmogal) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java]: yup, nvm. Just checked. :+1:

**Comment by bdmogal** [comment]
1 question, otherwise LGTM

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java (hunk 1 above)
Comment (poornachandra) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java]: Yes, it is okay to ignore the exception since the session is started as soon as the `executeStatement` is called. So, there is a high chance things will work. If for some reason session is not started, then later we'll throw exception anyways, and query will fail.

**Comment by bdmogal** [comment]
LGTM

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java [hunk 3]
```diff
@@ -86,7 +91,8 @@ protected CLIService createCLIService() {
   protected List<QueryResult> doFetchNextResults(OperationHandle handle, FetchOrientation fetchOrientation,
                                                  int size) throws Exception {
     Class cliServiceClass = Class.forName("org.apache.hive.service.cli.CLIService");
-    Method fetchResultsMethod = cliServiceClass.getMethod("fetchResults");
+    Method fetchResultsMethod = cliServiceClass.getMethod("fetchResults", OperationHandle.class,
+                                                          FetchOrientation.class, Long.TYPE);
```
Comment (gsps1) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java]: why this change?

Comment (poornachandra) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java]: It was broken for some reason, I fixed it.

**Comment by gsps1** [comment]
LGTM
